src/AspNet.Security.OAuth.Alipay/AlipayAuthenticationHandler.cs

```diff
@@ -0,0 +1,169 @@
+﻿using JetBrains.Annotations;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Text;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace AspNet.Security.OAuth.Alipay
+{
+    /// <summary>
+    /// 支付宝 OAUTH2.0 网站登录
+    /// 作者：Vicente.Yu
+    /// 创建时间：2019-12-12 00:35:35
+    /// QQ：8383463
+    /// </summary>
+    public class AlipayAuthenticationHandler : OAuthHandler<AlipayAuthenticationOptions>
+    {
+        private const string State = "state";
+        public AlipayAuthenticationHandler(
+            [NotNull] IOptionsMonitor<AlipayAuthenticationOptions> options,
+            [NotNull] ILoggerFactory logger,
+            [NotNull] UrlEncoder encoder,
+            [NotNull] ISystemClock clock)
+            : base(options, logger, encoder, clock) { }
+
+        protected override string BuildChallengeUrl(AuthenticationProperties properties, string redirectUri)
+        {
+            string stateValue = Options.StateDataFormat.Protect(properties);
+
+            var address = QueryHelpers.AddQueryString(Options.AuthorizationEndpoint, new Dictionary<string, string>
+            {
+                ["app_id"] = Options.ClientId,
+                ["scope"] = FormatScope(),
+                ["redirect_uri"] = redirectUri,
+                [State] = stateValue
```

Comment:
This constant isn't used anywhere else. Just remove it and use a literal like with the other 3 keys.